lib/Backend/arm64/LegalizeMD.cpp

```diff
@@ -526,15 +528,33 @@ void LegalizeMD::LegalizeLDIMM(IR::Instr * instr, IntConstType immed)
         if ((immed & 0xffff) == immed || (immed & 0xffff0000) == immed || (immed & 0xffff00000000ll) == immed || (immed & 0xffff000000000000ll) == immed)
         {
             instr->m_opcode = Js::OpCode::MOVZ;
+            uint32 shift = 0;
+            while ((immed & 0xffff) != immed)
```

Comment:
This loop is replicated several times. Maybe make it into a small inline helper?

Reply:
Sure, done.